### ISSUE TITLE
Provide a way for python to start an arrow draw but let the user finish it

### ISSUE THREAD

**Issue by db0** [octgn/OCTGN#1492]
To allow me to script the player requiring a target. This will be useful to scripting targeting ala hearthstone.

**Comment by brine** [comment]
why not just use the OnDoubleClick event?

**Comment by db0** [comment]
Because I want a visible way for the user to realize I'm expecting him to target one card with the effect of the other

**Comment by sIKE23** [comment]
I could find this useful too. I have effects in my game that I could trigger with this "start arrow" and let the player finish drawing the arrow to the intended target card.

**Comment by db0** [comment]
If you do that, provide a way for me to visibly point out what the effect they're targeting is for. E.g. a floating text box visible only until they land the arrow.

**Comment by brine** [comment]
some initial thoughts on this functionality:

the targeting arrow method in python is a mess.  While it looks nice visually, behind the scenes there's a lot of unintuitive stuff going on.  It's part of the reason why there are years-old issues still open regarding targetting arrows, and the whole thing needs to be redesigned before we can really do anything cool with them.  As a quick example, The only parts of the target arrow code that are stored within the card objects are IsTargeted (which is just a boolean), and TargetedBy (which lists the players currently targetting it).  Nowhere in the card data does it actually keep track of the start and end points of a target arrow.  The event trigger knows this only because it triggers off of the code that actually draws the arrow.

I do like the idea of a python call that gives users an easy way to choose cards on the table, but I'm not sure if the targeting arrows are the best way to do this.  I think the API call would need to freeze execution until the selections are made though, and return the selected cards as a list.

**Comment by db0** [comment]
Note that I'm not necessarily saying you should use the current arrow implementation. I'm OK With making something new from scratch that looks similar. I just posted it because you asked for API ideas ;)

**Comment by brine** [comment]
yea my usual method of brainstorming ideas involves breaking them apart first, so bear with me here ;)

**Comment by brine** [comment]
this is just an idea, not a permanent solution.  If you define a global variable to store the ID of a card that is requesting a target, then an OnDoubleClick event could check to see if that variable contains a card ID (else its null).  If yes, then process the script associated with that card.

To make it visually easy for the player to know which card is asking for the target, you could create a highlight around that card, or use the new card.filter() API to tint the card a brighter color so it stands out.

**Comment by Soul1355** [comment]
there's several issues I see with this. first, there's 2 ways I can think of to code the python part, 1) it freezes the python thread until the arrow is dropped (no remote calls, etc. likely to cause sync issues). Or, 2) it ends the python call, and uses an event trigger to pick back up after the arrow is dropped, which would get extremely complex having one event handle the result of any and every arrow. Second, it adds a lot of atypical behavior for the arrow, which creates a lot of opportunity for errors.

**Comment by brine** [comment]
we already have an event trigger when a card is drawn, all we would need the API to do is simulate the onMouseDown behavior so that OCTGN thinks that the player is holding down shift (the normal way to draw arrows)

**Comment by Soul1355** [comment]
yeah, I wasn't saying adding the event would be the problem, but I guess the complexity of what goes in that event isn't really my problem. 

But the way the arrow works doesn't support this and would have to be redone in order to do what your asking, and it's kinda hard for me to tell how much work that would be because it's code is kinda tied in with the code for dragging cards right now.
